### PR TITLE
Pubtools-sign-clearsign and pubtools-sign-containersign support

### PR DESCRIPTION
**Pull request by midnightercz**

### CODE REVIEW

File: src/pubtools/sign/results/signing_results.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from pubtools.sign.operations.base import SignOperation
+from pubtools.sign.results import SignerResults
+from pubtools.sign.results.sign_results import SignResults
+
+
+Signer = ForwardRef("Signer")
+
+
+@dataclasses.dataclass
+class SigningResults:
```
Comment (querti) [inline src/pubtools/sign/results/signing_results.py:16]: Could you rethink the naming of these classes? You have sign result, signing result, and signer result. Its confusing.

File: src/pubtools/sign/signers/msgsigner.py [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    status: str
+    error_message: str
+
+    def to_dict(self: SignerResults):
+        """Return dict representation of MsgSignerResults model."""
+        return {"status": self.status, "error_message": self.error_message}
+
+
+@dataclass()
+class MsgSigner(Signer):
```
Comment (querti) [inline src/pubtools/sign/signers/msgsigner.py:44]: Why is this a data class? It seems to be doing actual operations.

**Review by querti — CHANGES_REQUESTED** [review]

File: src/pubtools/sign/results/signing_results.py (hunk 1 above)
Comment (midnightercz) [reply src/pubtools/sign/results/signing_results.py:16]: would it be enough if I rename sign results to operation results?

File: src/pubtools/sign/signers/msgsigner.py (hunk 2 above)
Comment (querti) [reply src/pubtools/sign/signers/msgsigner.py:44]: Should every Python class just become dataclass then? My understanding of data classes is that they primarily exist to hold data and perform transformation operations on that data. If networking is involved then it's not appropriate to use them. PEP generally agrees: https://peps.python.org/pep-0557/#rationale .

File: src/pubtools/sign/results/signing_results.py (hunk 1 above)
Comment (querti) [reply src/pubtools/sign/results/signing_results.py:16]: I guess

File: src/pubtools/sign/signers/msgsigner.py (hunk 2 above)
Comment (midnightercz) [reply src/pubtools/sign/signers/msgsigner.py:44]: Where exactly in the PEP you see that? I see:
- The decorated classes are truly “normal” Python classes. The Data Class decorator should not interfere with any usage of the class.
-  One main design goal of Data Classes is to support static type checkers. 

There isn't any mention dataclasses should be use only to support model classes or any mention that they shouldn't be used to anything else than model classes. I used dataclass for the signer as it reads the configuration some input. It also enhance type checking. And for the last it allows you to skip init definition

Comment (querti) [reply src/pubtools/sign/signers/msgsigner.py:44]: "There have been numerous attempts to define classes which exist primarily to store values which are accessible by attribute lookup"

**Review by querti — APPROVED** [review]

**Pull request merged**